pkg/middlewares/aws_iam_policy_attachment_defaults_test.go

```diff
@@ -0,0 +1,91 @@
+package middlewares
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/r3labs/diff/v2"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+func TestAwsIamPolicyAttachmentDefaults_Execute(t *testing.T) {
+	tests := []struct {
+		name               string
+		remoteResources    []resource.Resource
+		resourcesFromState []resource.Resource
+		expected           []resource.Resource
+	}{
+		{
+			"test that default iam policy attachment are excluded when not managed by IaC",
+			[]resource.Resource{
+				&aws.AwsIamPolicyAttachment{
+					Id: "AWSServiceRoleForSSO-arn:aws:iam::aws:policy/aws-service-role/AWSSSOServiceRolePolicy",
+				},
+				&aws.AwsIamPolicyAttachment{
+					Id: "driftctl_test-arn:aws:iam::0123456789:policy/driftctl",
+				},
+			},
+			[]resource.Resource{
+				&aws.AwsIamPolicyAttachment{
+					Id: "driftctl_test-arn:aws:iam::0123456789:policy/driftctl",
+				},
+			},
+			[]resource.Resource{
+				&aws.AwsIamPolicyAttachment{
+					Id: "driftctl_test-arn:aws:iam::0123456789:policy/driftctl",
+				},
+			},
```

Comment:
Could you instead check for diffs in the changelog instead of checking the state ? Since the goal is to prevent adding noice in the changelog's diffs.

Reply:
Yes